app/src/main/java/com/teamltt/carcare/activity/TripsActivity.java

```diff
@@ -0,0 +1,179 @@
+ * limitations under the License.
+ */
+
+package com.teamltt.carcare.activity;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.ArrayAdapter;
+import android.widget.Spinner;
+
+import com.teamltt.carcare.R;
+import com.teamltt.carcare.database.DbHelper;
+import com.teamltt.carcare.fragment.MyObdResponseRecyclerViewAdapter;
+import com.teamltt.carcare.fragment.ObdResponseFragment;
+import com.teamltt.carcare.fragment.SimpleDividerItemDecoration;
+import com.teamltt.carcare.model.ObdContent;
+import com.teamltt.carcare.model.Trip;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TripsActivity extends AppCompatActivity implements ObdResponseFragment.OnListFragmentInteractionListener {
+
+    private static final String TAG = "TripsActivity";
+
+    private DbHelper dbHelper;
+
+    private Spinner spinner;
+    private Map<Trip, Long> tripLongMap;
+    private List<Trip> trips;
+    private ArrayAdapter<Trip> spinnerAdapter;
+
+    // Used to keep track of the items in the RecyclerView
+    private RecyclerView.Adapter responseListAdapter;
+    private List<ObdContent.ObdResponse> responses;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_trips);
+        spinner = (Spinner) findViewById(R.id.tripsSpinner);
+        tripLongMap = new HashMap<>();
+        trips = new ArrayList<>();
+        spinnerAdapter = new ArrayAdapter<>(this, android.R.layout.simple_spinner_item, trips);
+        spinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
+        spinner.setAdapter(spinnerAdapter);
+
+        responses = new ArrayList<>();
+        // Set up the list for responses
+        responseListAdapter = new MyObdResponseRecyclerViewAdapter(responses, this);
+        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.obd_reponse_list);
+        if (recyclerView != null) {
+            recyclerView.setHasFixedSize(true);
+            RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
+            recyclerView.setLayoutManager(layoutManager);
+            recyclerView.addItemDecoration(new SimpleDividerItemDecoration(this));
+            recyclerView.setAdapter(responseListAdapter);
+        }
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+        dbHelper = new DbHelper(this);
+        tripLongMap.clear();
+        trips.clear();
+        // TODO order these values chronologically or by key
```

Comment:
I think this `TODO` is completed with the call to `Collections.sort(trips)` and making the `Trip` class comparable